mws/apis/reports.py

```diff
@@ -10,6 +11,36 @@
 
 # TODO Add Schedule enumerations as constants
 
+def report_options_str(report_options):
+    """
+    Given a set of `report_options` as a dict, converts those options to a URL-encoded string.
+    
+    Each key-value pair in the dict is presented as "key=value", which is then URL-encoded
+    to, for instance, "key%3Dvalue".
+    Key-value pairs are then joined with ";".
+    
+    See examples in ReportType_Enumeration docs for details:
+    https://docs.developer.amazonservices.com/en_US/reports/Reports_ReportType.html
+    """
+    if not report_options:
+        return None
+    if not isinstance(report_options, dict):
+        raise ValueError("`report_options` must be a dict.")
+    output = []
+    for key, val in report_options.items():
+        out_val = val
+        if out_val is True or out_val is False:
+            # Value is explicitly `True` or `False`.
+            # The `is` identity check is necessary, otherwise `1 == True` and `0 == False`
+            # (both of which are accurate, because True and False can evaluate to ints 1 and 0).
+            # `True` and `False` must be output as a lowercase `"true"` and `"false"`, respectively.
+            out_val = str(out_val).lower()
+        # Use `urllib.parse.quote` to URL-encoded the output.
+        # (mostly just auto-convert the `=` to `%3D`, but might as well be safe).
+        output.append(urllib.parse.quote(f"{key}={out_val}"))
```

Comment:
probably a bigger conversation than for here, but f strings only appear in python 3.6 IIRC ?

Should we make this package python 3.6+ (current docs says 3.x)
Small thing - large implications!

Reply:
Noted, not worth breaking current versions at the moment.